### ISSUE TITLE
Edit Sloan plot

### ISSUE THREAD

**Issue by cliffbueno** [Asa12138/pctax#3]
Hi,
Thanks for the package. I used it to run the Sloan neutral model and make the plot. The default plot is great but the percentages in the upper left pie are overlapping and messy. It's because the neutral and below part are both very small proportions of my total OTUs so the slivers are smaller than the number width and they are right next to each other as well. Is there any way to tweak the pie percent number labels location?
Thanks,
Cliff

**Comment by Asa12138** [comment]
Hi,
Thanks for using `pctax` package.

The sloan plot is combined by two parts by `patchwork`, so it is not easy to edit plot by `ggplot2` grammar.

1. Adobe Illustrator 
I recommend using software such as Adobe Illustrator to edit the PDF directly, which is the most convenient.

2. pie_text_params
Use `pie_text_params` argument to pass some parameters to `geom_text`
```r
devtools::install_github("Asa12138/pctax")
plot(ncm_res,pie_text_params = list(col="red",size=1,nudge_x=0.5))
```

3. draw yourself
```r
  lib_ps("patchwork", library = FALSE)
  out <- ncm_res[[2]]
  mycols = c("Above" = "#069870", "Below" = "#e29e02", "In" = "#1e353a"
  lincol <- "#4a8fb8"
  p1 <- ggplot() +
    geom_line(data = out, aes(x = log(p), y = freq.pred), linewidth = 1.2, linetype = 1, col = lincol) +
    geom_line(data = out, aes(x = log(p), y = Lower), linewidth = 1.2, linetype = 2, col = lincol) +
    geom_line(data = out, aes(x = log(p), y = Upper), linewidth = 1.2, linetype = 2, col = lincol) +
    xlab("log10(mean relative abundance)") +
    ylab("Occurrence frequency")

    text_position <- c(log(max(out$p)) - 2, 0.05)

  p2 <- p1 +
    geom_point(data = out, aes(x = log(p), y = freq, color = group), size = 1) +
    scale_colour_manual(values = mycols) +
    annotate("text", x = text_position[1], y = text_position[2], label = paste("Nm = ", sprintf("%.0f", ncm_res[[1]][1] * ncm_res[[1]][3]), sep = ""), size = 4) +
    annotate("text", x = text_position[1], y = text_position[2] + 0.1, label = paste0("R2 = ", round(ncm_res[[1]][2], 3)), size = 4) +
    pctax_theme + theme(
      legend.position = c(0.85, 0.3),
      legend.title = element_blank(), legend.background = element_rect(I(0))
    )

  out$group %>%
    table() %>%
    as.data.frame() -> ad
  colnames(ad) <- c("type", "n")
  pie <- pcutils::gghuan(ad, name = FALSE, text_params = list(size = 2.5)) +
    xlim(0.2, 3.3) +
    scale_fill_manual(values = mycols) +
    theme(
      plot.background = element_rect(I(0), linetype = 0),
      panel.background = element_rect(I(0))
    ) # 去除图片背景白色

  p2 + patchwork::inset_element(pie, left = -0.1, bottom = 0.6, right = 0.3, top = 1.1)
```

**Comment by cliffbueno** [comment]
Awesome, thanks for the response and suggestions.

I'm trying to avoid editing the figures manually as they will inevitably have to be updated and I want this to be automated.

Both options 2 and 3 worked for me. I was able to change the text with the `pie_text_params` argument, so that is good to know about. But actually, I am going with option 3 as I am customizing even more things besides the pie text. In particular, I have two suggestions for you if you are interested. 

The first is to actually have the R2 written properly with a superscript. For that particular `annotate()` call I used the ^ combined the `parse = TRUE` option:
`annotate("text", 
x = text_position[1], 
y = text_position[2] + 0.1,
label = paste0("R^2 ==", round(ncm_res[[1]][2], 3)), size = 4, parse = TRUE)`

The second is to increase the size of the points in the legend. In the main plot the points have to be small because there are usually lots of points, but that doesn't mean they have to be small in the legend. I added a `guides()` call to increase the legend point size from 1 to 3:
`guides(color = guide_legend(override.aes = list(size = 3)))`

Thanks,
Cliff
